internal-enrichment/virustotal/src/virustotal/virustotal.py

```diff
@@ -91,104 +166,516 @@ def _create_yara_indicator(
+        json_data = self.client.get_domain_info(observable["observable_value"])
+        assert json_data
+        if "error" in json_data:
+            raise ValueError(json_data["error"]["message"])
+        elif "data" not in json_data or "attributes" not in json_data["data"]:
+            raise ValueError("An error has occurred.")
+
+        bundle_objects = []
+        attributes = json_data["data"]["attributes"]
+        malicious_count = attributes["last_analysis_stats"]["malicious"]
+        harmless_count = (
+            attributes["last_analysis_stats"]["harmless"]
+            + attributes["last_analysis_stats"]["undetected"]
+        )
+        dns_records = attributes["last_dns_records"]
+        score = round((malicious_count / (harmless_count + malicious_count)) * 100)
+        domain = observable["observable_value"]
+        now_time = datetime.datetime.utcnow()
+
+        # Create/attach external reference
+        external_reference = self.helper.api.external_reference.create(
+            source_name="VirusTotal",
+            url=f"https://www.virustotal.com/gui/domain/{domain}",
+            description="VirusTotal Report",
+        )
+        self.helper.api.stix_cyber_observable.add_external_reference(
+            id=observable["id"], external_reference_id=external_reference["id"]
+        )
+
+        # Update the score for the observable
+        self.helper.api.stix_cyber_observable.update_field(
+            id=observable["id"], input={"key": "x_opencti_score", "value": str(score)}
+        )
+
+        # Create IPv4 address observables for each A record
+        # And a Relationship between them and the observable
+        ip_addresses = [
+            record["value"] for record in dns_records if record["type"] == "A"
+        ]
+        for ip in ip_addresses:
+            # Have the use the API for these, see:
+            # https://github.com/OpenCTI-Platform/client-python/blob/master/examples/create_ip_domain_resolution.py
+            ipv4_stix = self.helper.api.stix_cyber_observable.create(
+                observableData={"type": "ipv4-addr", "value": ip}
+            )
+            self.helper.api.stix_cyber_observable_relationship.create(
+                fromId=observable["standard_id"],
+                toId=ipv4_stix["standard_id"],
+                createdBy=self.identity,
+                relationship_type="resolves-to",
+                update=True,
+                confidence=self.confidence_level,
+            )
+
+        # Create an Indicator if positive hits >= domain_indicator_create_positives specified in config
+        if (
+            self.domain_indicator_create_positives
+            and malicious_count >= self.domain_indicator_create_positives
+        ):
+
+            valid_until = now_time + datetime.timedelta(
+                minutes=self.domain_indicator_valid_minutes
+            )
+
+            indicator = self.helper.api.indicator.create(
```

Comment:
I saw that you ended up adding new entities via the API as well as adding objects to the bundle for them to later be written to the OpenCTI platform via the stix worker. I know that the code is not entirely written by you, but please follow this guideline as much as possible. https://luatix.notion.site/Connector-Development-06b2690697404b5ebc6e3556a1385940#1386fd230c534778a65fd3809959f656

If too many api create requests are sent off from the connector, the GraphQL db could end up blocking for a bit  https://github.com/OpenCTI-Platform/connectors/issues/433

Reply:
@nor3th Right, I typically will try and create the object for the Bundle first. In many cases, e.g. external references, resolves-to relationships, etc. this doesn't work, so I reverted to using the API.

Reply:
Sorry for the delay, I didn't manage to reply earlier. Could you please open an issue in the `client-python` repository for this problem? I'll release the PR for now. Thanks